src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -163,20 +231,29 @@ public function finishView(FormView $view, FormInterface $form, array $options)
      */
     public function configureOptions(OptionsResolver $resolver)
     {
-        $choiceListCache = & $this->choiceListCache;
+        $choiceListFactory = $this->choiceListFactory;
+
+        $choiceList = function (Options $options, $choiceList) use ($choiceListFactory) {
+            if (null !== $choiceList) {
+                trigger_error('The "choice_list" option is deprecated since version 2.7 and will be removed in 3.0. Use "choice_loader" instead.', E_USER_DEPRECATED);
```

Comment:
i don't think this will be triggered if people actually use the option as it will be overwritten. so I guess this needs to be done in a normalizer.